spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -290,7 +293,8 @@ public String toString() {
   private class BatchAppend extends BaseBatchWrite {
     @Override
     public void commit(WriterCommitMessage[] messages) {
-      AppendFiles append = table.newAppend();
+      AppendFiles append = table.newAppend().toBranch(branch);
```

Comment:
Rather than updating these individual cases, I think this should only modify `commitOperation`, similar to how the write-audit-publish support (using `stageOnly`) was added. It should go here: https://github.com/apache/iceberg/blob/master/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java#L212-L218

Reply:
There is also a `commitOperation` in `SparkPositionDeltaWrite` that should be updated as well.